### ISSUE TITLE
New warning 'invalid spec.yaml' when using external package

### ISSUE THREAD

**Issue by KineticTheory** [LLNL/spack#1914]
I'm seeing a new warning that is showing up when I use an external package:

```
==> Successfully installed superlu-dist
  Fetch: 0.03s.  Build: 42.64s.  Total: 42.68s.
[+] /yellow/usr/projects/draco/vendors/spack.ml.20161005/opt/spack/linux-redhat6-x86_64/intel-16.0.3/superlu-dist-4.3-kj66a5c6kvfljlkn6geshkt3dpytqa6y
==> Warning: Dependency missing due to corrupt install directory:
  /usr/projects/hpcsoft/toss2/common/intel-clusterstudio/2016.3.067/compilers_and_libraries_2016.3.210/linux/mkl
  Install prefix exists but contains no spec.yaml:
      /usr/projects/hpcsoft/toss2/common/intel-clusterstudio/2016.3.067/compilers_and_libraries_2016.3.210/linux/mkl
```

Because mkl is an external package, it will not have a `spec.yaml`.  I don't think spack should be issuing this warning.  

For completeness, my `packages.yaml` has the following:

```
packages:
  mkl:
    version: [11.3.3.210]
    paths:
      mkl@11.3.3.210%intel@16.0.3 arch=linux-redhat6-x86_64: /usr/projects/hpcsoft/toss2/common/intel-clusterstudio/2016.3.067/compilers_and_libraries_2016.3.210/linux/mkl
    buildable: False
  all:
    compiler: [intel@16.0.3]
    providers:
      blas: [mkl]
      lapack: [mkl]
```

I would like to know if there is something wrong with my configuration that is causing this warning.

**Comment by tgamblin** [comment]
Can you attach the results of `spack debug creadte-db-tarball`?

**Comment by KineticTheory** [comment]
@tgamblin 
[spack-db.develop.ac3e21f.2016-10-05-162749.tar.gz](https://github.com/LLNL/spack/files/512470/spack-db.develop.ac3e21f.2016-10-05-162749.tar.gz)

**Comment by pramodk** [comment]
I see the same issue!

**Comment by tgamblin** [comment]
Fixed by #2172: @KineticTheory can you verify?

**Comment by KineticTheory** [comment]
@tgamblin #2172 fixes #1914
